### PR TITLE
Porting commit f5986db184cf6d16dcd48760ac749a20236fb845

### PR DESCRIPTION
**Pull request by lamLeX**
This port fixes the bug  #6820 for helm3 which was fixed in helm2

### CODE REVIEW

**Comment by bacongobbler** [comment]
hi @lamLeX, a few things here:

- there's a typo in your commit message. "cheerypicked" should be "cherry-picked"
- if you truly wish to cherry-pick the fix in #4850, you should be able to run `git cherry-pick a9ca97b883f95e178843e8bd4d4d733af78ac48d` which should pick up the fix (as well as the original author of the code)
- in order to accept this PR, we require you to sign off on the commit to agree to the DCO. More details can be found in the failing check: https://github.com/helm/helm/pull/6821/checks?check_run_id=280173026
- it looks like the tests failed due to a code linting error. Would you mind fixing that?

**Comment by lamLeX** [comment]
@bacongobbler Thanks for your feedback. 
Due to the change in linting, I figured I should not do the cherry-picking but just one single commit.
However, if you prefer the cherry-picking path, I'm willing to split it into two commits, one for the cherry-picked commit and one for fixing the linting.

**Review by bacongobbler — APPROVED** [review]
thank you! Due to the release timing, Helm 3.0.0-rc.1 was just released. I'll consult with the other maintainers about seeing whether this should be cherry-picked in for the 3.0 release or it whether it should be pushed out to 3.1.0. In either case, thank you for the fix. :)

**Comment by bacongobbler** [comment]
As release maintainer, I think this warrants being cherry-picked into the 3.0 release as this is a feature regression from Helm 2. I'll cherry-pick this into the next RC. Thanks!

**Comment by lamLeX** [comment]
@bacongobbler Is there a way for me to know when this fix is picked? Will I have to wait till the release of 3.0.0 or would it be 3.0.0-rc.2 ? And is there any fixed timeline for it? I just want to know to plan my work accordingly.

**Comment by bacongobbler** [comment]
We will attach the `picked` label to this PR once it's been cherry-picked into the release branch. We are planning to cut the next release candidate today.

**Comment by lamLeX** [comment]
Awesome!!!

**Pull request merged**
